tools/buildings_to_dsm.py

```
@@ -277,56 +304,61 @@ def main(args):
+            # use the default scalar for point data
+            valuePass.SetInputComponentToProcess(0)
+            valuePass.SetInputArrayToProcess(vtk.VTK_SCALAR_MODE_USE_POINT_FIELD_DATA,
+                                             arrayName)
+            passes = vtk.vtkRenderPassCollection()
+            passes.AddItem(valuePass)
+            sequence = vtk.vtkSequencePass()
+            sequence.SetPasses(passes)
+            cameraPass = vtk.vtkCameraPass()
+            cameraPass.SetDelegatePass(sequence)
+            ren.SetPass(cameraPass)
+            # We have to render the points first, otherwise we get a segfault.
+            renWin.Render()
+            valuePass.SetInputArrayToProcess(vtk.VTK_SCALAR_MODE_USE_CELL_FIELD_DATA, arrayName)
+            renWin.Render()
+            elevationFlatVtk = valuePass.GetFloatImageDataArray(ren)
+            valuePass.ReleaseGraphicsResources(renWin)
+
+            print("Writing the DSM ...")
+            elevationFlat = numpy_support.vtk_to_numpy(elevationFlatVtk)
+            # VTK X,Y corresponds to numpy cols,rows. VTK stores arrays
+            # in Fortran order.
+            elevationTranspose = numpy.reshape(
+                elevationFlat, [dtm.RasterXSize, dtm.RasterYSize], "F")
+            # changes from cols, rows to rows,cols.
+            elevation = numpy.transpose(elevationTranspose)
+            # numpy rows increase as you go down, Y for VTK images increases as you go up
+            elevation = numpy.flip(elevation, 0)
+            if args.buildings_only:
+                dsmElevation = elevation
+            else:
+                # elevation has nans in places other than buildings
+                dsmElevation = numpy.fmax(dtmRaster, elevation)
+            dsm.GetRasterBand(1).WriteArray(dsmElevation)
+            if nodata:
+                dsm.GetRasterBand(1).SetNoDataValue(nodata)
+
+        except:
+            pass
 
```

Who throws the exception? Why pass and not print the error?

---

I wonder if this was just added for debugging before the offscreen rendering was fixed.  Maybe we can ignore this change.  It doesn't seem like a good idea to catch all exceptions and then pass without notification.